src/form-builder.tsx

```diff
@@ -1,18 +1,43 @@
-import { useEffect, useState } from "react";
+import { IFrontendEngineData } from "@lifesg/web-frontend-engine/components/types";
+import { forwardRef, useEffect, useImperativeHandle, useState } from "react";
 import { MainPanel, SidePanel } from "./components";
 import { Toasts } from "./components/common";
+import { ScreenNotSupportedErrorDisplay } from "./components/error-display/screen-not-supported-error";
 import { DisplayProvider } from "./context-providers";
-import { BuilderProvider } from "./context-providers/builder";
+import {
+    BuilderProvider,
+    TElementMap,
+    useBuilder,
+} from "./context-providers/builder";
 import { Container, ToastWrapper, Wrapper } from "./form-builder.styles";
-import { ScreenNotSupportedErrorDisplay } from "./components/error-display/screen-not-supported-error";
+import { generateSchema, translateSchema } from "./util/schema-translator";
+
+export interface IFormBuilderMethods {
+    generate?: (elementsList?: TElementMap) => IFrontendEngineData;
+    translate?: (text: string) => void;
```

Comment:
names can be more specific
```suggestion
    generateSchema?: (elementsList?: TElementMap) => IFrontendEngineData;
    parseSchema?: (schema: string) => void;
```